internal/archive/fsops.go

```diff
@@ -38,6 +38,12 @@ func Archive(sourcePath, targetArchivePath string, ignore []string) (e error) {
 		return err
 	}
 
+	// create folder of archive file if not exists
+	err = os.MkdirAll(filepath.Dir(targetArchivePath), os.ModePerm)
```

Comment:
You added this here, but you didn't remove the directory creation in the functions that call it (generateMtar in mtar.go and packModule in module_arch.go).
If you do remove it from packModule you should add it to copyModuleArchiveToResultDir.

Reply:
done